juju/juju
# Add controller config to the cache controller.

This branch adds controller config into the cache.

The branch also introduces additional change mechanisms to the modelcache worker. The modelcache worker is the thing that is responsible for populating the model cache. This worker now also listens to pubsub messages for updates. Just now it only listens to the controller config changed events, but it opens up for future additions.

This also introduces a finer grained change to the cache controller.

The controller name is only used now for the model summary watcher.

## QA steps

```sh
juju bootstrap
check the juju_engine_report to ensure workers started as expected
```
